src/components/views/rooms/MemberInfo.js

```diff
@@ -181,8 +181,8 @@ module.exports = withMatrixClient(React.createClass({
         }
     },
 
-    onRoomStateEvents: function(ev, state) {
-        this.forceUpdate();
+    onRoomMemberPowerlevel: function(ev, member) {
+        if (this.props.member.userId === member.userId) this.setState(this._calculateOpsPermissions(member));
```

Comment:
Yey for removing the `forceUpdate`. However, there is a subtlety here that you've missed (which is why `forceUpdate`s are so awful in the first place). 

In `render`, we get `poweLevelEvent` from the state of the room which is presumably partly why the `forceUpdate` is needed. So we need to re-render any time the room state (power level event) changes, and not just when a member's power updates. 

You might not have seen the failure mode because your `setState` would cause a re-render. 

There are two options:
 - Refactor such that we put `powerLevelEvent` in state, and have `onRoomStateEvents` update it (if it has actually changed) whilst also keeping your `onRoomMemberPowerLevel`.
 -  Use the `forceUpdate`.